modal-go/client_test.go

```diff
@@ -1,33 +1,91 @@
 package modal
 
 import (
+	"context"
 	"io"
 	"log/slog"
 	"os"
+	"sync"
 	"testing"
 
 	"github.com/onsi/gomega"
+	"google.golang.org/grpc"
 )
 
 func TestClientWithLogger(t *testing.T) {
 	g := gomega.NewWithT(t)
 
 	// Use a buffer to capture log output
 	r, w, err := os.Pipe()
-	g.Expect(err).To(gomega.BeNil())
+	g.Expect(err).ShouldNot(gomega.HaveOccurred())
 
 	logger := slog.New(slog.NewTextHandler(w, &slog.HandlerOptions{Level: slog.LevelDebug}))
 	g.Expect(logger).NotTo(gomega.BeNil())
 
 	client, err := NewClientWithOptions(&ClientParams{Logger: logger})
-	g.Expect(err).To(gomega.BeNil())
+	g.Expect(err).ShouldNot(gomega.HaveOccurred())
 	g.Expect(client).NotTo(gomega.BeNil())
 
 	w.Close()
 
 	output, err := io.ReadAll(r)
-	g.Expect(err).To(gomega.BeNil())
+	g.Expect(err).ShouldNot(gomega.HaveOccurred())
 
 	g.Expect(output).To(gomega.ContainSubstring("Initializing Modal client"))
 	g.Expect(output).To(gomega.ContainSubstring("Modal client initialized successfully"))
 }
+
+func TestClientWithCustomInterceptors(t *testing.T) {
+	g := gomega.NewWithT(t)
+
+	var firstCalled, secondCalled bool
+	var firstMethod, secondMethod string
+	var mu sync.Mutex
+
+	firstInterceptor := func(
+		ctx context.Context,
+		method string,
+		req, reply any,
+		cc *grpc.ClientConn,
+		invoker grpc.UnaryInvoker,
+		opts ...grpc.CallOption,
+	) error {
+		mu.Lock()
+		firstCalled = true
+		firstMethod = method
+		mu.Unlock()
```

Comment:
Very very nit:

```suggestion
		mu.Lock()
		defer mu.Unlock()
		firstCalled = true
		firstMethod = method
```

(It's slightly different behavior from what you have, but using `defer` feels more "go like")